tap_sftp/client.py

```diff
@@ -72,8 +73,11 @@ def sftp(self, sftp):
         self.__sftp = sftp
 
     def close(self):
-        self.sftp.close()
-        self.transport.close()
+        if hasattr(self, 'sftp') and self.sftp is not None:
```

Comment:
We need to check if the attribute exists here since if our attempt to connect (the `_attempt_connection` method above) fails the `sftp` attribute doesn't exist yet.